search_geo_arrayexpress.py

```diff
@@ -0,0 +1,155 @@
+    second_url = initial_url + "&RetMax=" + n_results + "&usehistory=y"
+    second_xml = requests.get(second_url)
+    second_dict = xmltodict.parse(second_xml.content)
+    query_key = second_dict['eSearchResult']['QueryKey']
+    webenv = second_dict['eSearchResult']['WebEnv']
+
+    # fetch returns up to fetch_retmax results, so we need to define the start
+    # position so we can increase the start position on subsequent fetches
+    retstart = 0
+
+    while retstart < int(n_results):
+        fetch_parameters = "&".join(
+          ["db=gds",
+           "query_key=" + query_key,
+           "WebEnv=" + webenv,
+           "retmax=10000",
+           "retstart=" + str(retstart)])
+        fetch_url = fetch_base + fetch_parameters
+        fetch_text = requests.get(fetch_url).text
+        for result in fetch_text.split("\n\n"):  # split by \n
+            record = result.split("\n")  # split by \n
+            for entry in record:
+                if entry in skip_these:  # data set should be skipped
+                    continue
+                else:  # otherwise, look for the line starting with "Platform"
+                    if entry.startswith("Platform"):  # parse 2nd last element
+                        n_samples = int(entry.split(" ")[-2])  # (n_samples)
+                        geo_dict[platform][1] += n_samples  # increment count
+        retstart += fetch_retmax  # increment the start position
+
+###############################################################################
+# ArrayExpress
+###############################################################################
+
+# set up search terms and dictionary to track n_samples
+# do not include GEO results in AE (directsub=on)
+ae_base_url = "https://www.ebi.ac.uk/arrayexpress/ArrayExpress-Experiments.txt?keywords="
+ae_array_url = ae_base_url + "&organism=Homo+sapiens&exptype%5B%5D=%22rna+assay%22&exptype%5B%5D=%22array+assay%22&array=&directsub=on"
+ae_rnaseq_url = ae_base_url + "&organism=Homo+sapiens&exptype%5B%5D=%22rna+assay%22&exptype%5B%5D=%22sequencing+assay%22&array=&directsub=on"
+
+ae_dict = {"array": [ae_array_url, 0],
+           "rnaseq": [ae_rnaseq_url, 0]}
+
+# for each platform in the dictionary, get results from url
+for platform in ae_dict:
+    url = ae_dict[platform][0]
+    results = requests.get(url).text.split("\n")[1:-1]  # skip first, last
+    for entry in results:
+        n_assays = int(entry.split('\t')[4])  # fifth column is n_assays
+        ae_dict[platform][1] += n_assays  # increment the count
+
+###############################################################################
+# Write results to output files
+###############################################################################
+
+total_array = geo_dict["array"][1] + ae_dict["array"][1]  # total number array
+total_rnaseq = geo_dict["rnaseq"][1] + ae_dict["rnaseq"][1]  # total n RNA-seq
+ratio = total_array/total_rnaseq  # array:RNA-seq
```

Comment:
If either `total` variable is 0 for whatever reason (internet or database fail), 137 _could_ fail with NaN math. I might add something between 136/137 - 
`assert (total_array != 0 and total_rnaseq != 0), "booooo zeroes"`